web/src/lib/components/map-page/map-settings-modal.svelte

```diff
@@ -27,9 +35,90 @@
 			Map Settings
 		</h1>
 
-		<form on:submit|preventDefault={() => dispatch('save', settings)} class="flex flex-col gap-4">
+		<form
+			on:submit|preventDefault={() => dispatch('save', settings)}
+			class="flex flex-col gap-4 text-immich-primary dark:text-immich-dark-primary"
+		>
 			<SettingSwitch title="Allow dark mode" bind:checked={settings.allowDarkMode} />
-			<SettingSwitch title="Show only favorites" bind:checked={settings.onlyFavorites} />
+			<SettingSwitch title="Only favorites" bind:checked={settings.onlyFavorites} />
+			{#if customDateRange}
+				<div in:fly={{ y: 10, duration: 200 }} class="flex flex-col gap-4">
+					<div class="flex justify-between items-center gap-8">
+						<label class="immich-form-label text-sm shrink-0" for="date-after">Date after</label>
+						<input
+							class="immich-form-input w-40"
+							type="date"
+							id="date-after"
+							max={settings.dateBefore}
+							bind:value={settings.dateAfter}
+						/>
+					</div>
+					<div class="flex justify-between items-center gap-8">
+						<label class="immich-form-label text-sm shrink-0" for="date-before">Date before</label>
+						<input
+							class="immich-form-input w-40"
+							type="date"
+							id="date-before"
+							bind:value={settings.dateBefore}
+						/>
+					</div>
+					<div class="flex justify-center text-xs">
+						<LinkButton
+							on:click={() => {
+								customDateRange = false;
+								settings.dateAfter = '';
+								settings.dateBefore = '';
+							}}
+						>
+							Remove custom date range
+						</LinkButton>
+					</div>
+				</div>
+			{:else}
+				<div in:fly={{ y: -10, duration: 200 }} class="flex flex-col gap-1">
+					<SettingSelect
+						label="Date range"
+						name="date-range"
+						bind:value={settings.relativeDate}
+						options={[
+							{
+								value: '',
+								text: '---'
```

Comment:
Would it make sense to put the text `All` here or similar?

Reply:
Yeah can do, the other values are also very subjective so feel free to suggest other date ranges

Reply:
What do you think about these:
- 24 hours
- 7 days
- 14 days
- 1 month
- 3 months
- 6 months
- 1 year
- All